### PR TITLE
BP-1085 Second Chance: Refactor

### PR DESCRIPTION
**Pull request by saranchuk-hys**

### CODE REVIEW

File: Controller/Redirect/Process.php [hunk 1]
```diff
@@ -353,6 +355,15 @@ public function execute()
     protected function handleFailed($statusCode)
     {
         $this->logger->addDebug(__METHOD__ . '|7|');
+
+        $this->eventManager->dispatch('buckaroo_process_handle_failed_before');
```
Comment (florinel-chis) [inline Controller/Redirect/Process.php:359]: does this require an event - or could have been handled via plugins (and check visibility of the method).

Comment (saranchuk-hys) [reply Controller/Redirect/Process.php:359]: yes it`s require event, on next step will be used in separate module

File: Controller/Payconiq/Process.php [hunk 2]
```diff
@@ -62,10 +61,10 @@ public function __construct(
         \Magento\Customer\Model\Customer $customerModel,
         \Magento\Customer\Model\ResourceModel\CustomerFactory $customerFactory,
         \Buckaroo\Magento2\Model\SecondChanceRepository $secondChanceRepository,
```
Comment (florinel-chis) [inline Controller/Payconiq/Process.php:63]: is this used?

File: Observer/SuccessOrder.php [hunk 3]
```diff
@@ -75,17 +75,6 @@ public function execute(\Magento\Framework\Event\Observer $observer)
     {
         $this->logging->addDebug(__METHOD__ . '|1|');
```
Comment (florinel-chis) [inline Observer/SuccessOrder.php:77]: secondChanceRepository - if this is not used, remove from constructor...

**Pull request merged**